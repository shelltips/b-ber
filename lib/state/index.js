'use strict';var _assign=require('babel-runtime/core-js/object/assign'),_assign2=_interopRequireDefault(_assign),_actions=require('./actions'),bberActions=_interopRequireWildcard(_actions);Object.defineProperty(exports,'__esModule',{value:!0});function _interopRequireWildcard(a){if(a&&a.__esModule)return a;var b={};if(null!=a)for(var c in a)Object.prototype.hasOwnProperty.call(a,c)&&(b[c]=a[c]);return b.default=a,b}function _interopRequireDefault(a){return a&&a.__esModule?a:{default:a}}var actions=(0,_assign2.default)({},bberActions);exports.default=actions;