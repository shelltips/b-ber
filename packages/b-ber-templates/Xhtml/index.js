<<<<<<< HEAD
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _vinyl = require('vinyl');

var _vinyl2 = _interopRequireDefault(_vinyl);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Xhtml = function () {
    function Xhtml() {
        (0, _classCallCheck3.default)(this, Xhtml);
    }

    (0, _createClass3.default)(Xhtml, null, [{
        key: 'head',
        value: function head() {

            // TODO: whitespace should be trimmed from the doc start and end after
            // b-ber runs

            return '<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n            <html xmlns="http://www.w3.org/1999/xhtml"\n                xmlns:epub="http://www.idpf.org/2007/ops"\n                xmlns:ibooks="http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0"\n                epub:prefix="ibooks: http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0">\n            <head>\n                <title></title>\n                <meta http-equiv="default-style" content="text/html charset=utf-8"/>\n                <!-- inject:css -->\n                <!-- end:css -->\n            </head>\n            <body>\n        ';
        }
    }, {
        key: 'body',
        value: function body() {
            return new _vinyl2.default({
                path: 'page.body.tmpl',
                contents: new Buffer('{% body %}')
            });
        }
    }, {
        key: 'tail',
        value: function tail() {
            return '\n                <!-- inject:js -->\n                <!-- end:js -->\n                <!-- inject:metadata -->\n                <!-- end:metadata -->\n            </body>\n        </html>\n        ';
        }
    }, {
        key: 'cover',
        value: function cover(_ref) {
            var width = _ref.width,
                height = _ref.height,
                href = _ref.href;

            return '\n            <section class="cover" style="text-align: center; padding: 0; margin: 0;">\n                <svg xmlns="http://www.w3.org/2000/svg" height="100%" preserveAspectRatio="xMidYMid meet" version="1.1" viewBox="0 0 ' + width + ' ' + height + '" width="100%" xmlns:xlink="http://www.w3.org/1999/xlink">\n                    <image width="' + width + '" height="' + height + '" xlink:href="../' + href + '"/>\n                </svg>\n            </section>\n        ';
        }
    }, {
        key: 'script',
        value: function script() {
            var type = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'application/javascript';
            var inline = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : false;

            return '<script type="' + type + '" ' + (inline ? '' : 'src="{% body %}"') + '></script>';
        }
    }, {
        key: 'stylesheet',
        value: function stylesheet() {
            return '<link rel="stylesheet" type="text/css" href="{% body %}"/>';
        }
    }, {
        key: 'loi',
        value: function loi() {
            return '\n            <section epub:type="loi" title="Figures" class="chapter figures">\n                <header>\n                    <h1>Figures</h1>\n                </header>\n            </section>\n        ';
        }
    }, {
        key: 'document',
        value: function document() {
            return new _vinyl2.default({
                path: 'xhtml.document.tmpl',
                contents: new Buffer(Xhtml.head() + '\n                    {% body %}\n                ' + Xhtml.tail() + '\n            ')
            });
        }
    }]);
    return Xhtml;
}();

exports.default = Xhtml;
||||||| parent of 832bbc7... Restructuring
=======
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _classCallCheck2 = require('babel-runtime/helpers/classCallCheck');

var _classCallCheck3 = _interopRequireDefault(_classCallCheck2);

var _createClass2 = require('babel-runtime/helpers/createClass');

var _createClass3 = _interopRequireDefault(_createClass2);

var _vinyl = require('vinyl');

var _vinyl2 = _interopRequireDefault(_vinyl);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Xhtml = function () {
    function Xhtml() {
        (0, _classCallCheck3.default)(this, Xhtml);
    }

    (0, _createClass3.default)(Xhtml, null, [{
        key: 'head',
        value: function head() {
            return '\n            <?xml version="1.0" encoding="UTF-8" standalone="no"?>\n            <html xmlns="http://www.w3.org/1999/xhtml"\n                xmlns:epub="http://www.idpf.org/2007/ops"\n                xmlns:ibooks="http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0"\n                epub:prefix="ibooks: http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0">\n            <head>\n                <title></title>\n                <meta http-equiv="default-style" content="text/html charset=utf-8"/>\n                <!-- inject:css -->\n                <!-- end:css -->\n            </head>\n            <body>\n        ';
        }
    }, {
        key: 'body',
        value: function body() {
            return new _vinyl2.default({
                path: 'page.body.tmpl',
                contents: new Buffer('{% body %}')
            });
        }
    }, {
        key: 'tail',
        value: function tail() {
            return '\n                <!-- inject:js -->\n                <!-- end:js -->\n                <!-- inject:metadata -->\n                <!-- end:metadata -->\n            </body>\n        </html>\n        ';
        }
    }, {
        key: 'cover',
        value: function cover(_ref) {
            var width = _ref.width,
                height = _ref.height,
                href = _ref.href;

            return '\n            <section class="cover" style="text-align: center; padding: 0; margin: 0;">\n                <svg xmlns="http://www.w3.org/2000/svg" height="100%" preserveAspectRatio="xMidYMid meet" version="1.1" viewBox="0 0 ' + width + ' ' + height + '" width="100%" xmlns:xlink="http://www.w3.org/1999/xlink">\n                    <image width="' + width + '" height="' + height + '" xlink:href="../' + href + '"/>\n                </svg>\n            </section>\n        ';
        }
    }, {
        key: 'script',
        value: function script() {
            var type = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'application/javascript';

            return '<script type="' + type + '" src="{% body %}"></script>';
        }
    }, {
        key: 'stylesheet',
        value: function stylesheet() {
            return '<link rel="stylesheet" type="text/css" href="{% body %}"/>';
        }
    }, {
        key: 'document',
        value: function document() {
            return new _vinyl2.default({
                path: 'xhtml.document.tmpl',
                contents: new Buffer('\n                ' + Xhtml.head() + '\n                    {% body %}\n                ' + Xhtml.tail() + '\n            ')
            });
        }
    }]);
    return Xhtml;
}();

exports.default = Xhtml;

/*
const loiLeader = () =>
    `<section epub:type="loi" title="Figures" class="chapter figures">
        <header>
            <h1>Figures</h1>
        </header>
    </section>`

export {pageHead, pageTail, pageBody, page, loiLeader}
*/
>>>>>>> 832bbc7... Restructuring
