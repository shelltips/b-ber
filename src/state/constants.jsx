
/* eslint-disable import/prefer-default-export */

export const SET_BBER = 'SET_BBER'
export const GET_BBER = 'GET_BBER'
