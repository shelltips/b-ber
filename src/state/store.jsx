
export default { pages: [], images: [], build: null, bber: {} }
