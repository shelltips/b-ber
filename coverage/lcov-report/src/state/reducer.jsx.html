<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/state/reducer.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">src/state</a> reducer.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.44% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">20% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">44.44% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/18</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&nbsp;
import * as actionType from './constants'
import initialState from './store'
&nbsp;
let state = initialState
&nbsp;
f</span>unction <span class="fstat-no" title="function not covered" >_get({ props }) <span class="cstat-no" title="statement not covered" ></span>{
  c</span>onst args = <span class="cstat-no" title="statement not covered" >Array.prototype.slice.call(props, 0)
  c</span>onst res = <span class="cstat-no" title="statement not covered" >{}
  l</span>et <span class="cstat-no" title="statement not covered" >prop
&nbsp;
<span class="cstat-no" title="statement not covered" >  while ((prop = args.pop())) {</span>
<span class="cstat-no" title="statement not covered" >    if ({}.hasOwnProperty.call(state, prop)) {</span>
<span class="cstat-no" title="statement not covered" >      res[prop] = state[prop]</span>
    }
  }
<span class="cstat-no" title="statement not covered" >  return res</span>
}
&nbsp;
const reducer = (props) =&gt; {
  const { type } = props
  switch (type) {
    case actionType.SET_BBER:
      state = Object.assign({}, state, props, { lastActionType: type })
      return state
<span class="branch-1 cbranch-no" title="branch not covered" >    case actionType.GET_BBER:</span>
<span class="cstat-no" title="statement not covered" >      return _get(props)</span>
<span class="branch-2 cbranch-no" title="branch not covered" >    default:</span>
<span class="cstat-no" title="statement not covered" >      return state</span>
  }
}
&nbsp;
export default reducer
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="http://istanbul-js.org/" target="_blank">istanbul</a> at Fri Mar 17 2017 11:59:53 GMT+0100 (CET)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
