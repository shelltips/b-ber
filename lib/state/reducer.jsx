
import * as actionType from './constants'
import initialState from './store'

let state = initialState

function _get({ props }) {
  const args = Array.prototype.slice.call(props, 0)
  const res = {}
  let prop

  while ((prop = args.pop())) {
    if ({}.hasOwnProperty.call(state, prop)) {
      res[prop] = state[prop]
    }
  }
  return res
}

const reducer = (props) => {
  const { type } = props
  switch (type) {
    case actionType.SET_BBER:
      state = Object.assign({}, state, props, { lastActionType: type })
      return state
    case actionType.GET_BBER:
      return _get(props)
    default:
      return state
  }
}

export default reducer
