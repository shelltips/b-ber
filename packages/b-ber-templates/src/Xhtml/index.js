import File from 'vinyl'
import state from '@canopycanopycanopy/b-ber-lib/State'

class Xhtml {
    static head() {
        // TODO: whitespace should be trimmed from the doc start and end after
        // b-ber runs

        const robotsMeta = state.config.private
            ? '<meta name="robots" content="noindex,nofollow"/>'
            : '<meta name="robots" content="index,follow"/>'
        return `<?xml version="1.0" encoding="UTF-8" standalone="no"?>
            <html xmlns="http://www.w3.org/1999/xhtml"
                xmlns:epub="http://www.idpf.org/2007/ops"
                xmlns:ibooks="http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0"
                epub:prefix="ibooks: http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0">
            <head>
                <title></title>
                <meta http-equiv="default-style" content="text/html charset=utf-8"/>
                ${robotsMeta}
                <!-- inject:css -->
                <!-- end:css -->
            </head>
            <body>
        `
    }
    static body() {
        return new File({
            path: 'page.body.tmpl',
            contents: new Buffer('{% body %}'),
        })
    }
    static tail() {
        return `
                <!-- inject:js -->
                <!-- end:js -->
                <!-- inject:metadata -->
                <!-- end:metadata -->
            </body>
        </html>
        `
    }
    static cover({ width, height, href }) {
        return `
            <section class="cover" style="text-align: center; padding: 0; margin: 0;">
                <svg xmlns="http://www.w3.org/2000/svg" height="100%" preserveAspectRatio="xMidYMid meet" version="1.1" viewBox="0 0 ${width} ${height}" width="100%" xmlns:xlink="http://www.w3.org/1999/xlink">
                    <image width="${width}" height="${height}" xlink:href="../${href}"/>
                </svg>
            </section>
        `
    }
    static script(type = 'application/javascript', inline = false) {
        return inline
            ? `<script type="${type}">{% body %}</script>`
            : `<script type="${type}" src="{% body %}"></script>`
    }
    static stylesheet() {
        return '<link rel="stylesheet" type="text/css" href="{% body %}"/>'
    }

    static loi() {
        return `
            <section epub:type="loi" title="Figures" class="chapter figures">
                <header>
                    <h1>Figures</h1>
                </header>
            </section>
        `
    }

    static document() {
        return new File({
            path: 'xhtml.document.tmpl',
            contents: new Buffer(`${Xhtml.head()}
                    {% body %}
                ${Xhtml.tail()}
            `),
        })
    }
}

export default Xhtml
