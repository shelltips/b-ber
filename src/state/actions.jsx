
/* eslint-disable import/prefer-default-export */

import * as actionType from './constants'
import reducer from './reducer'

export function setBber(props) {
  return reducer({ type: actionType.SET_BBER, ...props })
}

export function getBber(...props) {
  return reducer({ type: actionType.GET_BBER, props })
}
