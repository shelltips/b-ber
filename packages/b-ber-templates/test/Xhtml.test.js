import state from '@canopycanopycanopy/b-ber-lib/State'
import Xhtml from '../src/Xhtml'

describe('templates.Xhtml', () => {
    it('creates a head element', () => {

        state.config.private = true
        expect(Xhtml.head()).toMatchSnapshot()

        state.config.private = false
        expect(Xhtml.head()).toMatchSnapshot()
    })

    it('creates a body element', () => {
        expect(Xhtml.body().contents).toMatchSnapshot()
        expect(Xhtml.body().history).toMatchSnapshot()
        expect(Xhtml.tail()).toMatchSnapshot()
    })

    it('creates a cover page', () => {
        expect(Xhtml.cover({width: 1, height: 1, href: 'test'})).toMatchSnapshot()
    })

    it('creates a script element', () => {
        expect(Xhtml.script()).toMatchSnapshot()
    })

    it('creates as link element', () => {
        expect(Xhtml.stylesheet()).toMatchSnapshot()
    })

    it('creates the loi elements', () => {
        expect(Xhtml.loi()).toMatchSnapshot()
    })

    it('creates an html document', () => {
        expect(Xhtml.document().contents).toMatchSnapshot()
        expect(Xhtml.document().history).toMatchSnapshot()
    })

})
