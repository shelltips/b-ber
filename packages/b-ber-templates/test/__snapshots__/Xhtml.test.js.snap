// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`templates.Xhtml creates a body element 1`] = `
Object {
  "data": Array [
    123,
    37,
    32,
    98,
    111,
    100,
    121,
    32,
    37,
    125,
  ],
  "type": "Buffer",
}
`;

exports[`templates.Xhtml creates a body element 2`] = `
Array [
  "page.body.tmpl",
]
`;

exports[`templates.Xhtml creates a body element 3`] = `
"
                <!-- inject:js -->
                <!-- end:js -->
                <!-- inject:metadata -->
                <!-- end:metadata -->
            </body>
        </html>
        "
`;

exports[`templates.Xhtml creates a cover page 1`] = `
"
            <section class=\\"cover\\" style=\\"text-align: center; padding: 0; margin: 0;\\">
                <svg xmlns=\\"http://www.w3.org/2000/svg\\" height=\\"100%\\" preserveAspectRatio=\\"xMidYMid meet\\" version=\\"1.1\\" viewBox=\\"0 0 1 1\\" width=\\"100%\\" xmlns:xlink=\\"http://www.w3.org/1999/xlink\\">
                    <image width=\\"1\\" height=\\"1\\" xlink:href=\\"../test\\"/>
                </svg>
            </section>
        "
`;

exports[`templates.Xhtml creates a head element 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?>
            <html xmlns=\\"http://www.w3.org/1999/xhtml\\"
                xmlns:epub=\\"http://www.idpf.org/2007/ops\\"
                xmlns:ibooks=\\"http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0\\"
                epub:prefix=\\"ibooks: http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0\\">
            <head>
                <title></title>
                <meta http-equiv=\\"default-style\\" content=\\"text/html charset=utf-8\\"/>
                <meta name=\\"robots\\" content=\\"noindex,nofollow\\"/>
                <!-- inject:css -->
                <!-- end:css -->
            </head>
            <body>
        "
`;

exports[`templates.Xhtml creates a head element 2`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" standalone=\\"no\\"?>
            <html xmlns=\\"http://www.w3.org/1999/xhtml\\"
                xmlns:epub=\\"http://www.idpf.org/2007/ops\\"
                xmlns:ibooks=\\"http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0\\"
                epub:prefix=\\"ibooks: http://vocabulary.itunes.apple.com/rdf/ibooks/vocabulary-extensions-1.0\\">
            <head>
                <title></title>
                <meta http-equiv=\\"default-style\\" content=\\"text/html charset=utf-8\\"/>
                <meta name=\\"robots\\" content=\\"index,follow\\"/>
                <!-- inject:css -->
                <!-- end:css -->
            </head>
            <body>
        "
`;

exports[`templates.Xhtml creates a script element 1`] = `"<script type=\\"application/javascript\\" src=\\"{% body %}\\"></script>"`;

exports[`templates.Xhtml creates an html document 1`] = `
Object {
  "data": Array [
    60,
    63,
    120,
    109,
    108,
    32,
    118,
    101,
    114,
    115,
    105,
    111,
    110,
    61,
    34,
    49,
    46,
    48,
    34,
    32,
    101,
    110,
    99,
    111,
    100,
    105,
    110,
    103,
    61,
    34,
    85,
    84,
    70,
    45,
    56,
    34,
    32,
    115,
    116,
    97,
    110,
    100,
    97,
    108,
    111,
    110,
    101,
    61,
    34,
    110,
    111,
    34,
    63,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    104,
    116,
    109,
    108,
    32,
    120,
    109,
    108,
    110,
    115,
    61,
    34,
    104,
    116,
    116,
    112,
    58,
    47,
    47,
    119,
    119,
    119,
    46,
    119,
    51,
    46,
    111,
    114,
    103,
    47,
    49,
    57,
    57,
    57,
    47,
    120,
    104,
    116,
    109,
    108,
    34,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    120,
    109,
    108,
    110,
    115,
    58,
    101,
    112,
    117,
    98,
    61,
    34,
    104,
    116,
    116,
    112,
    58,
    47,
    47,
    119,
    119,
    119,
    46,
    105,
    100,
    112,
    102,
    46,
    111,
    114,
    103,
    47,
    50,
    48,
    48,
    55,
    47,
    111,
    112,
    115,
    34,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    120,
    109,
    108,
    110,
    115,
    58,
    105,
    98,
    111,
    111,
    107,
    115,
    61,
    34,
    104,
    116,
    116,
    112,
    58,
    47,
    47,
    118,
    111,
    99,
    97,
    98,
    117,
    108,
    97,
    114,
    121,
    46,
    105,
    116,
    117,
    110,
    101,
    115,
    46,
    97,
    112,
    112,
    108,
    101,
    46,
    99,
    111,
    109,
    47,
    114,
    100,
    102,
    47,
    105,
    98,
    111,
    111,
    107,
    115,
    47,
    118,
    111,
    99,
    97,
    98,
    117,
    108,
    97,
    114,
    121,
    45,
    101,
    120,
    116,
    101,
    110,
    115,
    105,
    111,
    110,
    115,
    45,
    49,
    46,
    48,
    34,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    101,
    112,
    117,
    98,
    58,
    112,
    114,
    101,
    102,
    105,
    120,
    61,
    34,
    105,
    98,
    111,
    111,
    107,
    115,
    58,
    32,
    104,
    116,
    116,
    112,
    58,
    47,
    47,
    118,
    111,
    99,
    97,
    98,
    117,
    108,
    97,
    114,
    121,
    46,
    105,
    116,
    117,
    110,
    101,
    115,
    46,
    97,
    112,
    112,
    108,
    101,
    46,
    99,
    111,
    109,
    47,
    114,
    100,
    102,
    47,
    105,
    98,
    111,
    111,
    107,
    115,
    47,
    118,
    111,
    99,
    97,
    98,
    117,
    108,
    97,
    114,
    121,
    45,
    101,
    120,
    116,
    101,
    110,
    115,
    105,
    111,
    110,
    115,
    45,
    49,
    46,
    48,
    34,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    104,
    101,
    97,
    100,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    116,
    105,
    116,
    108,
    101,
    62,
    60,
    47,
    116,
    105,
    116,
    108,
    101,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    109,
    101,
    116,
    97,
    32,
    104,
    116,
    116,
    112,
    45,
    101,
    113,
    117,
    105,
    118,
    61,
    34,
    100,
    101,
    102,
    97,
    117,
    108,
    116,
    45,
    115,
    116,
    121,
    108,
    101,
    34,
    32,
    99,
    111,
    110,
    116,
    101,
    110,
    116,
    61,
    34,
    116,
    101,
    120,
    116,
    47,
    104,
    116,
    109,
    108,
    32,
    99,
    104,
    97,
    114,
    115,
    101,
    116,
    61,
    117,
    116,
    102,
    45,
    56,
    34,
    47,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    109,
    101,
    116,
    97,
    32,
    110,
    97,
    109,
    101,
    61,
    34,
    114,
    111,
    98,
    111,
    116,
    115,
    34,
    32,
    99,
    111,
    110,
    116,
    101,
    110,
    116,
    61,
    34,
    105,
    110,
    100,
    101,
    120,
    44,
    102,
    111,
    108,
    108,
    111,
    119,
    34,
    47,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    33,
    45,
    45,
    32,
    105,
    110,
    106,
    101,
    99,
    116,
    58,
    99,
    115,
    115,
    32,
    45,
    45,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    33,
    45,
    45,
    32,
    101,
    110,
    100,
    58,
    99,
    115,
    115,
    32,
    45,
    45,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    47,
    104,
    101,
    97,
    100,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    98,
    111,
    100,
    121,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    123,
    37,
    32,
    98,
    111,
    100,
    121,
    32,
    37,
    125,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    33,
    45,
    45,
    32,
    105,
    110,
    106,
    101,
    99,
    116,
    58,
    106,
    115,
    32,
    45,
    45,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    33,
    45,
    45,
    32,
    101,
    110,
    100,
    58,
    106,
    115,
    32,
    45,
    45,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    33,
    45,
    45,
    32,
    105,
    110,
    106,
    101,
    99,
    116,
    58,
    109,
    101,
    116,
    97,
    100,
    97,
    116,
    97,
    32,
    45,
    45,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    33,
    45,
    45,
    32,
    101,
    110,
    100,
    58,
    109,
    101,
    116,
    97,
    100,
    97,
    116,
    97,
    32,
    45,
    45,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    47,
    98,
    111,
    100,
    121,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    60,
    47,
    104,
    116,
    109,
    108,
    62,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    10,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
    32,
  ],
  "type": "Buffer",
}
`;

exports[`templates.Xhtml creates an html document 2`] = `
Array [
  "xhtml.document.tmpl",
]
`;

exports[`templates.Xhtml creates as link element 1`] = `"<link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"{% body %}\\"/>"`;

exports[`templates.Xhtml creates the loi elements 1`] = `
"
            <section epub:type=\\"loi\\" title=\\"Figures\\" class=\\"chapter figures\\">
                <header>
                    <h1>Figures</h1>
                </header>
            </section>
        "
`;
